tests/ui/coherence/occurs-check/opaques.rs

```diff
@@ -0,0 +1,37 @@
+//revisions: old next
+//[next] compile-flags: -Ztrait-solver=next
+
+// A regression test for #105787
+
+//[old] known-bug: #105787
```

Comment:
we do not emit projection obligations for opaque types because the old solver does not support them :grin: so this is still an issue in the old solver.